includes/Core/Modules/Module.php

```diff
@@ -319,15 +320,15 @@ public function is_connected() {
 				continue;
 			}
 
-			if ( empty( $datapoint_services[ $datapoint ] ) ) {
+			if ( empty( $datapoint_services[ $datapoint_name ]['service'] ) ) {
 				continue;
 			}
 
-			if ( ! isset( $service_batches[ $datapoint_services[ $datapoint ] ] ) ) {
-				$service_batches[ $datapoint_services[ $datapoint ] ] = $this->google_services[ $datapoint_services[ $datapoint ] ]->createBatch();
+			if ( ! isset( $service_batches[ $datapoint_services[ $datapoint_name ]['service'] ] ) ) {
```

Comment:
Let's extract the service name to make this a bit more readable.
E.g.
```php
$datapoint_service = $datapoint_definitions[ "{$data->method}:{$data->datapoint}" ]['service'];
```

Reply:
Updated.